pkg/activator/net/revision_backends.go

```diff
@@ -226,13 +230,39 @@ func (rw *revisionWatcher) checkDests(dests sets.String) {
 	if len(dests) == 0 {
 		// We must have scaled down.
 		rw.clusterIPHealthy = false
+		rw.healthyPods = sets.NewString()
 		rw.logger.Debug("ClusterIP is no longer healthy.")
 		// Send update that we are now inactive (both params invalid).
 		rw.sendUpdate("", nil)
 		return
 	}
 
-	// First check the clusterIP. We can't cache it, since user might go rogue
+	// If we have discovered that this revision cannot be probed directly
+	// do not spend time trying.
+	if !rw.noPodAddressability {
```

Comment:
nit: double negative is difficult to understand

Reply:
But that is how the thought goes: "we're not in pod addressable mode" :)

Reply:
I agree, why not do `podsAddressable` and flip the conditions?

Reply:
I think the code is much less brittle that way, since we now *must explicitly* initialize the flag to true 😿